### PR TITLE
[Silabs] Configure Light-Switch as a LIT ICD app

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description
PR configures the Light-Swith as a LIT ICD with Dynamic Sit Lit Support
- Update ZAP file with the necessary configs
- Add Common ICD shell that is enabled if the device is an ICD and shell is enabled
- Clean defines checks
- Moves shell files into their respective directory
- Adds a LIT ICD Light-Switch Zap file to avoid causing build issues or changing the behavior of other platforms sample apps.

PR also pulls latest docker image which reducdes the size of the efr32 docker image.

#### Tests
Manual tests to validate everything works as expected

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

File: src/lib/shell/BUILD.gn [hunk 1]
```diff
@@ -57,9 +58,14 @@ static_library("shell") {
   } else if (chip_device_platform == "efr32" ||
              chip_device_platform == "SiWx917") {
     sources += [
+      # Temporary workaround to prevent a circular dependecy while restructuring the silabs shell files
+      "${chip_root}/examples/platform/silabs/shell/MatterShell.h",
       "MainLoopSilabs.cpp",
       "streamer_silabs.cpp",
     ]
+
+    # Temporary workaround to prevent a circular dependecy while restructuring the silabs shell files
```
Comment (jmartinez-silabs) [inline src/lib/shell/BUILD.gn:67]: Can we track this in a internal ticket please

File: examples/light-switch-app/light-switch-common/BUILD.gn [hunk 2]
```diff
@@ -14,8 +14,13 @@
 
 import("//build_overrides/chip.gni")
 import("${chip_root}/src/app/chip_data_model.gni")
+import("${chip_root}/src/app/icd/icd.gni")
 
 chip_data_model("light-switch-common") {
-  zap_file = "light-switch-app.zap"
+  if (chip_enable_icd_lit) {
+    zap_file = "lit-icd-light-switch-app.zap"
```
Comment (andy31415) [inline examples/light-switch-app/light-switch-common/BUILD.gn:21]: should the name order be consistent? one is lit-icd and the config is icd_lit

File: examples/platform/silabs/BaseApplication.cpp [hunk 3]
```diff
@@ -37,9 +37,13 @@
 #endif // QR_CODE_ENABLED
 #endif // DISPLAY_ENABLED
 
-#if CHIP_CONFIG_ENABLE_ICD_SERVER == 1
+#if defined(CHIP_CONFIG_ENABLE_ICD_SERVER) && CHIP_CONFIG_ENABLE_ICD_SERVER
```
Comment (andy31415) [inline examples/platform/silabs/BaseApplication.cpp:40]: We should not use defined/ifdef.

Please include relevant headers and only use `#if` and `#if !`.

**Review by andy31415 — CHANGES_REQUESTED** [review]
Changes requested: do not use `ifdef` or `defined(....)`

File: src/lib/shell/BUILD.gn [hunk 4]
```diff
@@ -57,9 +58,15 @@ static_library("shell") {
...
     sources += [
+      # Temporary workaround to prevent a circular dependecy while restructuring the silabs shell files
+      # Due to some odd dependency issue, we need the source file and the include_dir for the build to pass.
+      "${chip_root}/examples/platform/silabs/shell/MatterShell.h",
       "MainLoopSilabs.cpp",
       "streamer_silabs.cpp",
     ]
+
+    # Temporary workaround to prevent a circular dependecy while restructuring the silabs shell files
+    include_dirs = [ "${chip_root}/examples/platform/silabs/shell/" ]
```
Comment (andy31415) [inline src/lib/shell/BUILD.gn:69]: Please describe the circular dependency. Generally this is not ok: we cannot link from src into examples. You may need to fix this as a prerequisite before this PR can go in.

File: src/lib/shell/BUILD.gn [hunk 5]
```diff
@@ -57,9 +58,15 @@ static_library("shell") {
   } else if (chip_device_platform == "efr32" ||
              chip_device_platform == "SiWx917") {
     sources += [
+      # Temporary workaround to prevent a circular dependecy while restructuring the silabs shell files
+      # Due to some odd dependency issue, we need the source file and the include_dir for the build to pass.
+      "${chip_root}/examples/platform/silabs/shell/MatterShell.h",
```
Comment (andy31415) [inline src/lib/shell/BUILD.gn:63]: This does not look ok. Saying in a comment "I know it does not look ok" does not make it better.

What kind of a commitment can we get for this to get fixed? I would approve if I am allowed to roll the PR back if not fixed within X time where X is some reasonably picked but short timeframe, like 2-4 weeks.

**Comment by andy31415** [comment]
> Changes requested: do not use `ifdef` or `defined(....)`

Also making lib depend on examples is something that needs fixing.

File: examples/light-switch-app/light-switch-common/BUILD.gn (hunk 2 above)
Comment (mkardous-silabs) [reply examples/light-switch-app/light-switch-common/BUILD.gn]: done

File: examples/platform/silabs/BaseApplication.cpp (hunk 3 above)
Comment (mkardous-silabs) [reply examples/platform/silabs/BaseApplication.cpp]: done

File: src/lib/shell/BUILD.gn (hunk 4 above)
Comment (mkardous-silabs) [reply src/lib/shell/BUILD.gn]: reverted these changes until i have time to do the full cleanup.

File: src/lib/shell/BUILD.gn (hunk 5 above)
Comment (mkardous-silabs) [reply src/lib/shell/BUILD.gn]: reverted these changes until i have time to do the full cleanup.

**Review by yunhanw-google — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**